vllm/distributed/kv_transfer/kv_connector/v1/nixl_connector.py

```diff
@@ -700,7 +705,12 @@ def _nixl_handshake_listener(
         logger.debug("Starting listening on path: %s", path)
         with zmq_ctx(zmq.ROUTER, path) as sock:
             ready_event.set()
-            while True:
+            poller = zmq.Poller()
+            poller.register(sock, zmq.POLLIN)
+            while not stop_event.is_set():
+                events = dict(poller.poll(timeout=poll_timeout * 1000))
+                if sock not in events:
+                    continue
```

Comment:
ok so this replaces the busy waiting mechanism without hanging on recv so we can quit, nice